.github/workflows/build-lint-test.yml

```diff
@@ -0,0 +1,41 @@
+# This workflow will install Python dependencies, run tests and lint on Python 3.11.
+# This initiates on every push or pull_request to develop
+
+name: Build, Lint, and Test
+
+on:
+  push:
+    branches: [ "develop" ]
+  pull_request:
+    branches: [ "develop" ]
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v4
+    - name: Set up Python 3.11
+      uses: actions/setup-python@v3
+      with:
+        python-version: "3.11"
```

Comment:
No worries, I have implemented this. The version flow is as follows:

1. The Python version can be specified using the workflow_dispatch input python-version, on manual runs.
2. If not specified, it will be read from pyproject.toml, using the lowest compatibility.
3. In case of no specification found, will default to 3.11.